src/modeler/main.cpp

```diff
@@ -64,13 +61,13 @@ class SystemLinearProblem
         }
 
         LinearProblemBuilder linear_problem_builder(fillers_ptr);
-        LinearProblemData data(dataSeriesRepo);
         FillContext dummy_time_scenario_ctx = {parameters.firstTimeStep, parameters.lastTimeStep};
-        linear_problem_builder.build(pb, data, dummy_time_scenario_ctx);
+        linear_problem_builder.build(pb, dataSeries_, dummy_time_scenario_ctx);
     }
 
 private:
     const Study::SystemModel::System& system_;
+    ILinearProblemData& dataSeries_;
```

Comment:
Do we really need a non const reference ?
Can't we do with a copy or a const reference ?
Since data series is only used in Provide, maybe it make more sens to move dataSeries as a parameter of build.
SystemLinearProblem is not much of a system rather than a SystemLinearProblemBuilder